From 9ce487ab1db5d4b2e1d22e8b8436c4750c309719 Mon Sep 17 00:00:00 2001
From: Sam McLeod <not@telling.github.com>
Date: Sat, 8 Aug 2026 19:29:27 +1000
Subject: [PATCH 7/8] HID: hid-ids: add 2026 Steam Controller device IDs

Four device IDs used by the 2026 Steam Controller (codename Triton),
taken from Valve's linux-integration tree at tag 6.18.42-valve2:

  IBEX     0x1302  controller, wired
  IBEX_BLE 0x1303  controller, Bluetooth
  PROTEUS  0x1304  wireless puck
  NEREID   0x1305

Inserted rather than taking Valve's hid-ids.h wholesale: their 6.18
header predates the Harmonix and PDP RiffMaster IDs that 7.2 added, so
copying it whole removes definitions other in-tree HID drivers need.

Origin: Valve Software, linux-integration 6.18.42-valve2 (af6356cf2488)
---
 drivers/hid/hid-ids.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/hid/hid-ids.h b/drivers/hid/hid-ids.h
index 1059922baaac..b3559e70eec8 100644
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -1374,6 +1374,10 @@
 #define USB_DEVICE_ID_STEAM_CONTROLLER		0x1102
 #define USB_DEVICE_ID_STEAM_CONTROLLER_WIRELESS	0x1142
 #define USB_DEVICE_ID_STEAM_DECK	0x1205
+#define USB_DEVICE_ID_STEAM_CONTROLLER_IBEX	0x1302
+#define USB_DEVICE_ID_STEAM_CONTROLLER_IBEX_BLE	0x1303
+#define USB_DEVICE_ID_STEAM_CONTROLLER_PROTEUS	0x1304
+#define USB_DEVICE_ID_STEAM_CONTROLLER_NEREID	0x1305
 
 #define USB_VENDOR_ID_STEELSERIES	0x1038
 #define USB_DEVICE_ID_STEELSERIES_SRWS1	0x1410
-- 
2.50.1

